internal/config/variables/variables.go

```diff
@@ -0,0 +1,531 @@
+package variables
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/ext/typeexpr"
+	"github.com/hashicorp/hcl/v2/gohcl"
+	"github.com/hashicorp/hcl/v2/hclsyntax"
+	hcljson "github.com/hashicorp/hcl/v2/json"
+	pb "github.com/hashicorp/waypoint/internal/server/gen"
+	"github.com/zclconf/go-cty/cty"
+	"github.com/zclconf/go-cty/cty/convert"
+	"github.com/zclconf/go-cty/cty/gocty"
+)
+
+var (
+	// Variable value sources
+	// listed in descending precedence order for ease of reference
+	sourceCLI     = "cli"
+	sourceFile    = "file"
+	sourceEnv     = "env"
+	sourceVCS     = "vcs"
+	sourceServer  = "server"
+	sourceDefault = "default"
+
+	// Prefix for collecting variable values from environment variables
+	varEnvPrefix = "WP_VAR_"
```

Comment:
all these above here probably want to be `const`, and not var.

Reply:
done